File: VBWebApp/Controllers/dbo_Table_3Controller.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Data.SqlClient
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports VBWebApp
Imports VBWebApp.Data

Public Class dbo_Table_3Controller
    Inherits System.Web.Mvc.Controller

    Private db As New VBWebAppContext

    ' GET: dbo_Table_3
    Function Index() As ActionResult

        Dim reqparam = Request.Params
        Dim requestbody = Request.InputStream

        'Dim buf(1024) As Byte
        'requestbody.Read(buf, 0, 1024)

        Dim lst As List(Of dbo_Table_3) = db.Database.SqlQuery(Of dbo_Table_3)("SELECT Aaaa FROM dbo_Table_3").ToList()

        Return View(db.dbo_Table_3.ToList())
    End Function

    ' GET: dbo_Table_3/Details/5
    Function Details(ByVal id As String) As ActionResult
        If IsNothing(id) Then
            Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
        End If
        Dim dbo_Table_3 As dbo_Table_3 = db.dbo_Table_3.Find(id)
        If IsNothing(dbo_Table_3) Then
            Return HttpNotFound()
        End If
        Return View(dbo_Table_3)
    End Function

    ' GET: dbo_Table_3/Create
    Function Create() As ActionResult
        Return View()
    End Function

    ' POST: dbo_Table_3/Create
    '過多ポスティング攻撃を防止するには、バインド先とする特定のプロパティを有効にしてください。
    '詳細については、https://go.microsoft.com/fwlink/?LinkId=317598 をご覧ください。
    <HttpPost()>
    <ValidateAntiForgeryToken()>
    Function Create(<Bind(Include:="Aaaa")> ByVal dbo_Table_3 As dbo_Table_3) As ActionResult
        If ModelState.IsValid Then
            db.dbo_Table_3.Add(dbo_Table_3)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End If
        Return View(dbo_Table_3)
    End Function

    ' GET: dbo_Table_3/Edit/5
    Function Edit(ByVal id As String) As ActionResult
        If IsNothing(id) Then
            Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
        End If

        Dim Sql As String = "SELECT Aaaa FROM dbo_Table_3 where Aaaa=@Aaaa"
        Dim param As SqlParameter = New SqlParameter("@Aaaa", id)
        Dim rs = db.Database.SqlQuery(Of dbo_Table_3)(Sql, param)
        Dim rec As dbo_Table_3 = rs.First()


        Dim dbo_Table_3 As dbo_Table_3 = db.dbo_Table_3.Find(id)
        If IsNothing(dbo_Table_3) Then
            Return HttpNotFound()
        End If
        Return View(dbo_Table_3)
    End Function

    ' POST: dbo_Table_3/Edit/5
    '過多ポスティング攻撃を防止するには、バインド先とする特定のプロパティを有効にしてください。
    '詳細については、https://go.microsoft.com/fwlink/?LinkId=317598 をご覧ください。
    <HttpPost()>
    <ValidateAntiForgeryToken()>
    Function Edit(<Bind(Include:="Aaaa")> ByVal dbo_Table_3 As dbo_Table_3) As ActionResult
        If ModelState.IsValid Then
            db.Entry(dbo_Table_3).State = EntityState.Modified
            db.SaveChanges()
            Return RedirectToAction("Index")
        End If
        Return View(dbo_Table_3)
    End Function

    ' GET: dbo_Table_3/Delete/5
    Function Delete(ByVal id As String) As ActionResult
        If IsNothing(id) Then
            Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
        End If
        Dim dbo_Table_3 As dbo_Table_3 = db.dbo_Table_3.Find(id)
        If IsNothing(dbo_Table_3) Then
            Return HttpNotFound()
        End If
        Return View(dbo_Table_3)
    End Function

    ' POST: dbo_Table_3/Delete/5
    <HttpPost()>
    <ActionName("Delete")>
    <ValidateAntiForgeryToken()>
    Function DeleteConfirmed(ByVal id As String) As ActionResult
        Dim dbo_Table_3 As dbo_Table_3 = db.dbo_Table_3.Find(id)
        db.dbo_Table_3.Remove(dbo_Table_3)
        db.SaveChanges()
        Return RedirectToAction("Index")
    End Function

    Protected Overrides Sub Dispose(ByVal disposing As Boolean)
        If (disposing) Then
            db.Dispose()
        End If
        MyBase.Dispose(disposing)
    End Sub
End Class

File: WebApplication2/Controllers/dbo_Table_3Controller.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Mvc;
using Microsoft.AspNetCore.Mvc.Rendering;
using Microsoft.EntityFrameworkCore;
using CSWebApp.Data;
using CSWebApp.Models;
using Microsoft.Data.SqlClient;
using System.Diagnostics;

namespace CSWebApp.Controllers
{
    public class dbo_Table_3Controller : Controller
    {
        private readonly CSWebAppContext _context;

        public dbo_Table_3Controller(CSWebAppContext context)
        {
            _context = context;
        }

        // GET: dbo_Table_3
        public async Task<IActionResult> Index()
        {

            //リクエストパラメータ
            IQueryCollection reqparams = Request.Query;

            //post body
            Stream postbody = Request.Body;

            var rs2 = _context.Database.SqlQuery<string>($"SELECT DB_NAME() AS Value");
            string db = rs2.First<string>();
            Trace.WriteLine("current db:" + db);

            rs2 = _context.Database.SqlQuery<string>($"SELECT CURRENT_USER AS Value");
            string user = rs2.First<string>();
            Trace.WriteLine("current user:" + user);

            rs2 = _context.Database.SqlQuery<string>($"SELECT SCHEMA_NAME() AS Value");
            string schema = rs2.First<string>();
            Trace.WriteLine("current schema:" + schema);

            rs2 = _context.Database.SqlQuery<string>($"SELECT @@SERVICENAME AS Value");
            string servicename = rs2.First<string>();
            Trace.WriteLine("current servicename:" + servicename);

            List<dbo_Table_3> lst = await _context.dbo_Table_3.ToListAsync();

            FormattableString sql = $"SELECT Aaaa FROM dbo_Table_3";
            var convert = _context.Database.SqlQuery<dbo_Table_3>(sql);
            lst = convert.ToList();

            return View(lst);

        }

        // GET: dbo_Table_3/Details/5
        public async Task<IActionResult> Details(string id)
        {
            if (id == null)
            {
                return NotFound();
            }

            var dbo_Table_3 = await _context.dbo_Table_3
                .FirstOrDefaultAsync(m => m.Aaaa == id);
            if (dbo_Table_3 == null)
            {
                return NotFound();
            }

            string sql = @"SELECT Aaaa FROM dbo_Table_3 where Aaaa=@Aaaa";
            object[] param = new object[] { new SqlParameter("@Aaaa", id) };
            var rs = _context.Database.SqlQueryRaw<dbo_Table_3>(sql,param);
            dbo_Table_3 = rs.First<dbo_Table_3>();


            return View(dbo_Table_3);
        }

        // GET: dbo_Table_3/Create
        public IActionResult Create()
        {
            return View();
        }

        // POST: dbo_Table_3/Create
        // To protect from overposting attacks, enable the specific properties you want to bind to.
        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<IActionResult> Create([Bind("Aaaa")] dbo_Table_3 dbo_Table_3)
        {
            if (ModelState.IsValid)
            {
                _context.Add(dbo_Table_3);
                await _context.SaveChangesAsync();
                return RedirectToAction(nameof(Index));
            }
            return View(dbo_Table_3);
        }

        // GET: dbo_Table_3/Edit/5
        public async Task<IActionResult> Edit(string id)
        {
            if (id == null)
            {
                return NotFound();
            }

            var dbo_Table_3 = await _context.dbo_Table_3.FindAsync(id);
            if (dbo_Table_3 == null)
            {
                return NotFound();
            }
            return View(dbo_Table_3);
        }

        // POST: dbo_Table_3/Edit/5
        // To protect from overposting attacks, enable the specific properties you want to bind to.
        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
        [HttpPost]
        [ValidateAntiForgeryToken]
        public async Task<IActionResult> Edit(string id, [Bind("Aaaa")] dbo_Table_3 dbo_Table_3)
        {
            if (id != dbo_Table_3.Aaaa)
            {
                return NotFound();
            }

            if (ModelState.IsValid)
            {
                try
                {
                    _context.Update(dbo_Table_3);
                    await _context.SaveChangesAsync();
                }
                catch (DbUpdateConcurrencyException)
                {
                    if (!dbo_Table_3Exists(dbo_Table_3.Aaaa))
                    {
                        return NotFound();
                    }
                    else
                    {
                        throw;
                    }
                }
                return RedirectToAction(nameof(Index));
            }
            return View(dbo_Table_3);
        }

        // GET: dbo_Table_3/Delete/5
        public async Task<IActionResult> Delete(string id)
        {
            if (id == null)
            {
                return NotFound();
            }

            var dbo_Table_3 = await _context.dbo_Table_3
                .FirstOrDefaultAsync(m => m.Aaaa == id);
            if (dbo_Table_3 == null)
            {
                return NotFound();
            }

            return View(dbo_Table_3);
        }

        // POST: dbo_Table_3/Delete/5
        [HttpPost, ActionName("Delete")]
        [ValidateAntiForgeryToken]
        public async Task<IActionResult> DeleteConfirmed(string id)
        {
            var dbo_Table_3 = await _context.dbo_Table_3.FindAsync(id);
            if (dbo_Table_3 != null)
            {
                _context.dbo_Table_3.Remove(dbo_Table_3);
            }

            await _context.SaveChangesAsync();
            return RedirectToAction(nameof(Index));
        }

        private bool dbo_Table_3Exists(string id)
        {
            return _context.dbo_Table_3.Any(e => e.Aaaa == id);
        }
    }
}
